cmd/nodepool/core/create.go

```diff
@@ -184,3 +191,72 @@ func validateHostedClusterPayloadSupportsNodePoolCPUArch(ctx context.Context, cl
 
 	return nil
 }
+
+// validMinorVersionCompatibility validates that the NodePool version is compatible with the HostedCluster version.
+// For 4.even versions, it allows y-2 difference.
+// For 4.odd versions, it allows y-1 difference.
+// NodePool version cannot be higher than control plane version.
+func validMinorVersionCompatibility(ctx context.Context, client crclient.Client, name, namespace, nodePoolReleaseImage string) error {
+	if nodePoolReleaseImage == "" {
+		return nil
+	}
+	logger := ctrl.LoggerFrom(ctx)
+
+	hcluster := &hyperv1.HostedCluster{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, hcluster); err != nil {
+		// This is expected to happen when we create a cluster since there is no created HostedCluster CR to check the
+		// payload from.
+		logger.Info("WARNING: failed to get HostedCluster to check version compatibility")
+		return nil
+	}
+
+	// Parse control plane version
+	controlPlaneVersion, err := semver.Parse(hcluster.Status.Version.Desired.Version)
+	if err != nil {
+		return fmt.Errorf("parsing control plane version (%s): %w", hcluster.Status.Version.Desired.Version, err)
+	}
+
+	pullSecret := &corev1.Secret{}
+	if err = client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: hcluster.Spec.PullSecret.Name}, pullSecret); err != nil {
+		return fmt.Errorf("failed to get pull secret: %w", err)
+	}
+	releaseProvider := &releaseinfo.RegistryClientProvider{}
+	releaseImage, err := releaseProvider.Lookup(ctx, nodePoolReleaseImage, pullSecret.Data[corev1.DockerConfigJsonKey])
+	if err != nil {
+		// Skip version check in disconnected environment where registry access is not available
+		logger.Info("WARNING: Unable to access the payload, skipping the Minor Version check.", "error", err.Error())
+		return nil
+	}
+
+	// Parse NodePool version
+	nodePoolVersion, err := semver.Parse(releaseImage.Version())
+	if err != nil {
+		return fmt.Errorf("parsing NodePool version (%s): %w", releaseImage.Version(), err)
+	}
+
+	// NodePool version cannot be higher than control plane version
+	if nodePoolVersion.Minor > controlPlaneVersion.Minor {
+		return fmt.Errorf("NodePool minor version %d.%d cannot be higher than the HostedCluster minor version %d.%d",
```

Comment:
HyperShift's position on patch-version skew is [unclear][1], but in the absence of explicit docs allowing 4.y.(z > z') NodePools on 4.y.z' HostedClusters, you may want to also have a guard for that:

```go
else if nodePoolVersion.Minor == controlPlaneVersion.Minor && nodePoolVersion.Patch > controlPlaneVersion.Patch {
```

Or if HyperShift's patch-skew policy is "go for it; we test and support that!" then hooray, you're fine here without trying to check that situation :)

[1]: https://issues.redhat.com/browse/RFE-6410

Reply:
done